docs/architecture/grpc-for-wcf-developers/self-hosted.md

```diff
@@ -1,11 +1,185 @@
 ---
 title: Self-hosted gRPC applications - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: Deploying ASP.NET Core gRPC applications as self-hosted services
 author: markrendle
 ms.date: 09/02/2019
 ---
 
 # Self-hosted gRPC applications
 
+Although ASP.NET Core 3.0 applications can be hosted in IIS on Windows Server, at present, it is not possible to host a gRPC application in IIS because some of the HTTP/2 functionality is not yet supported. This functionality is expected in a future update to Windows Server.
+
+You can run your application as a Windows Service, or as a Linux service controlled by [systemd](https://en.wikipedia.org/wiki/Systemd), thanks to some new features in the .NET Core 3.0 hosting extensions.
+
+## Running your app as a Windows service
+
+To configure your ASP.NET Core application to run as a Windows service, install the [Microsoft.Extensions.Hosting.WindowsServices](https://www.nuget.org/packages/Microsoft.Extensions.Hosting.WindowsServices) package from NuGet. Then add a call to `UseWindowsService` to the `CreateHostBuilder` method in `Program.cs`.
+
+```csharp
+public static IHostBuilder CreateHostBuilder(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .UseWindowsService() // Enable running as a Windows service
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+            webBuilder.UseStartup<Startup>();
+        });
+```
+
+> [!NOTE]
+> If the application is not running as a Windows service, the `UseWindowsService` method doesn't do anything.
+
+Now publish your application for the relevant Windows runtime (e.g. `win-x64`), either from Visual Studio by right-clicking the project and choosing *Publish* from the context menu, or from the dotnet CLI using the following command.
+
+```console
+dotnet publish -c Release -r win-x64 -o ./publish
```

Comment:
Why self-contained?

Reply:
Why not?

Reply:
Breaks servicing. Not the biggest deal

Reply:
Added a paragraph on FD and SC and a link to https://docs.microsoft.com/en-us/dotnet/core/deploying/ for more info.